cumulusci/tasks/salesforce/package_upload.py

```diff
@@ -59,6 +70,43 @@ def _run_task(self):
             self._set_dependencies()
             self._log_package_upload_success()
 
+    """This functions validates the major version and minor version if passed and sets them in the options dictionary if not passed."""
+
+    def _validate_versions(self):
+
+        """Fetching the latest major,minor and patch version from Database"""
+        version = self._get_one_record(
+            (
+                "SELECT MajorVersion,"
+                "MinorVersion,"
+                "PatchVersion,ReleaseState FROM MetadataPackageVersion Order by MajorVersion Desc,MinorVersion Desc, PatchVersion Desc LIMIT 1"
+            ),
+            "Version not found",
+        )
+
+        """This if-else condition updates the major version to latest major version in options if not passed via command line and validates it if passed. """
+        if "major_version" in self.options:
+            if int(self.options["major_version"]) < version["MajorVersion"]:
```

Comment:
The conversion `int()` can raise `ValueError` if the supplied value isn't a number. You could add an exception handler here to catch that and re-raise a `TaskOptionsError`.